test/api/requests/build.test.js

```diff
@@ -623,6 +623,127 @@ describe('Build API', () => {
       .catch(done);
     });
 
+    it('should update the publishedAt from processing -> success status', (done) => {
+      let siteId;
+      let buildId;
+      const sitePromise = factory.site();
+
+      Promise.props({
+        site: sitePromise,
+        build: factory.build({
+          site: sitePromise,
+          commitSha,
+          state: 'processing',
+          startedAt: new Date(),
+        }),
+      })
+      .then((promisedValues) => {
+        expect(promisedValues.site.publishedAt).to.be.null;
+        siteId = promisedValues.site.id;
+        buildId = promisedValues.build.id;
+
+        return postBuildStatus({
+          build: promisedValues.build,
+          status: 'success',
+          message: '',
+        });
+      })
+      .then(() => Site.findByPk(siteId))
+      .then((site) => {
+        expect(site.publishedAt).to.be.a('date');
+        return Build.findByPk(buildId);
+      })
+      .then((build) => {
```

Comment:
It looks like testing the effects on the build are handled in a different spec (`should mark a build successful`), consider omitting them here or more explicitly combining them into one spec.